tutorials/26_Hybrid_Retrieval.ipynb

```diff
@@ -0,0 +1,451 @@
+    "\n",
+    "We use a [re-ranker based on a cross-encoder](https://docs.haystack.deepset.ai/docs/ranker#sentencetransformersranker) that scores the relevancy of all candidates for the given search query.\n",
+    "For more information about the `Ranker`, check the Haystack [docs](https://docs.haystack.deepset.ai/docs/ranker)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "d_RiKspTlXRl"
+   },
+   "outputs": [],
+   "source": [
+    "from haystack.nodes import JoinDocuments, SentenceTransformersRanker\n",
+    "\n",
+    "join_documents = JoinDocuments(join_mode=\"concatenate\")\n",
+    "rerank = SentenceTransformersRanker(model_name_or_path=\"cross-encoder/ms-marco-MiniLM-L-6-v2\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "PexSrsBLlXRp"
+   },
+   "source": [
+    "## 5) Create an Hybrid Retrieval Pipeline\n",
+    "\n",
+    "With a Haystack `Pipeline`, you can connect your building blocks into a search pipeline. Under the hood, `Pipelines` are Directed Acyclic Graphs (DAGs) that you can easily customize for your own use cases. To speed things up, Haystack also comes with a few predefined `Pipelines`.\n",
+    "You can learn more about Pipelines in the [docs](https://docs.haystack.deepset.ai/docs/pipelines)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "i0XLbnAXlXRt"
+   },
+   "outputs": [],
+   "source": [
+    "from haystack.pipelines import Pipeline\n",
+    "\n",
+    "pipeline = Pipeline()\n",
+    "pipeline.add_node(component=sparse_retriever, name=\"SparseRetriever\", inputs=[\"Query\"])\n",
+    "pipeline.add_node(component=dense_retriever, name=\"DenseRetriever\", inputs=[\"Query\"])\n",
+    "pipeline.add_node(component=join_documents, name=\"JoinDocuments\", inputs=[\"SparseRetriever\", \"DenseRetriever\"])\n",
+    "pipeline.add_node(component=rerank, name=\"ReRanker\", inputs=[\"JoinDocuments\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "V3bsFkHuhHn4"
+   },
+   "source": [
+    "### Generating a Pipeline Diagram\n",
+    "\n",
+    "With any Pipeline, whether prebuilt or custom constructed, you can save a diagram showing how all the components are connected."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "id": "oCIMtwmThQG4"
+   },
+   "outputs": [],
+   "source": [
+    "# Uncomment the following to generate the images\n",
+    "# !apt install libgraphviz-dev\n",
+    "# !pip install pygraphviz\n",
+    "\n",
+    "# pipeline.draw(\"pipeline_hybrid.png\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "sTTVLUJylXRx"
+   },
+   "source": [
+    "Search an article with Hybrid Retrieval. If you want to see all the steps, enable `debug=True` in `JoinDocuments`'s `params`."
```

Comment:
You can add a new section titled as `## Trying Out the Hybrid Pipeline` here covering pipeline.run and printing parts

Reply:
done